scripts/release/msigdb-gsea/build_gsea.pl

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+our $VERSION = 1.0;
+
+use autodie qw/:all/;
+use Capture::Tiny ':all';
+use Cwd;
+use English qw/-no_match_vars/;
+use Getopt::Long;
+use Readonly;
+
+Readonly my $DEFAULT_REPO_VERSION => 'master';
+my ($pathway_exchange_repo_version, $curator_tool_repo_verison, $help);
+GetOptions(
+    'pathway-exchange-version:s' => \$pathway_exchange_repo_version,
+    'curator-tool-version:s' => \$curator_tool_repo_verison,
+    'help' => \$help,
+);
+
+if ($help) {
+    print usage_instructions();
+    exit;
+}
+
+$pathway_exchange_repo_version ||= $DEFAULT_REPO_VERSION;
+$curator_tool_repo_verison ||= $DEFAULT_REPO_VERSION;
+
+my %repositories = (
+    'Pathway-Exchange' => {
+        url => 'https://github.com/reactome/Pathway-Exchange.git',
+        version => $pathway_exchange_repo_version,
+        ant_xml_file => 'GSEAdeployAsApplication.xml',
+        output => 'GSEAExport.jar',
+    },
+    'CuratorTool' => {
+        url => 'https://github.com/reactome/CuratorTool.git',
+        version => $curator_tool_repo_verison,
+        ant_xml_file => 'WebELVDiagram.xml',
+        output => 'WebELVTool/*.jar',
+    }
+);
+
+# Build required output from each repository
+foreach my $repository_name (keys %repositories) {
+    my $current_dir = getcwd;
+    my $repo_version = $repositories{$repository_name}{'version'};
+    # Pull latest changes if local repository exists
+    if (-d "$repository_name/.git") {
+        chdir $repository_name;
+        my $stderr = run_command('git pull');
+        if (trim($stderr)) {
+            print STDERR "Problem pulling $repository_name:\n$stderr";
+        }
+
+        chdir $current_dir;
+    # Clone repository if it doesn't exist locally
+    } else {
+        my $repository_url = $repositories{$repository_name}{'url'};
+        my $stderr = run_command("git clone $repository_url", {
+            ignore_error => qr/^Cloning into .*/
```

Comment:
Yeah, it drives me crazy that certain git command output goes to standard error :).  I think it's because standard error is also sometimes used by convention for something that isn't desired in standard output even though it may not be an error.

I'd like to keep standard error only for true errors, though, so a non-zero size error file is a signal that a problem occurred.

Reply:
So what your telling me is what we need is another stream type. We could have 'standard error', 'standard out' and 'standard whatever'

Reply:
I don't know if it's possible to create another stream for shell commands.  The mixing of info and errors in standard error has come up a couple of times, but it's been really infrequent, specific, and small messages that need to be taken out.  So far, I've just been filtering standard error when it happens.